src/components/Buttons/ButtonsAdmin/ButtonsAdmin.jsx

```diff
@@ -1,28 +1,164 @@
+import React from 'react';
 import PropTypes from 'prop-types';
 import { NavLink } from 'react-router-dom';
 import { Button } from '../Button/Button';
-import { BUTTONS_NAMES, BUTTONS_TYPES } from '../../../shared/constants';
-
-export function ButtonsAdmin({ id }) {
-  return (
-    <>
-      <NavLink to={`/tasks/${id}`}>
-        <Button title={BUTTONS_NAMES.tasks} />
-      </NavLink>
-      <NavLink to={`/progress/${id}`}>
-        <Button title={BUTTONS_NAMES.progress} />
-      </NavLink>
-
-      <Button title={BUTTONS_NAMES.edit} stylingType={BUTTONS_TYPES.typeEdit} />
-      <Button title={BUTTONS_NAMES.delete} stylingType={BUTTONS_TYPES.typeDelete} />
-    </>
-  );
+import { BUTTONS_NAMES, BUTTONS_TYPES, USER_ROLES } from '../../../shared/constants';
+import { DeleteForm } from '../../Forms/DeleteForm/DeleteForm';
+import { MyModal } from '../../Common/Modal/Modal';
+import { getAllUsers, getUserData, removeUserData } from '../../../services/users-services ';
+import noop from '../../../shared/noop';
+import style from './ButtonsAdmin.module.css';
+import { CreateMemberForm } from '../../Forms/CreateMemberForm/CreateMemberForm';
+import { AuthContext } from '../../../Hooks/useAuth';
+
+export class ButtonsAdminMemberPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isDeleteModalOpen: false,
+      isEditModalOpen: false,
+      userData: {},
+    };
+  }
+
+  async getUserData() {
+    const { id, toggleError } = this.props;
+    const userData = await getUserData(id);
+    if (userData) {
+      this.setState({ userData });
+    } else {
+      toggleError();
+    }
+  }
+
+  deleteUserHandler = async () => {
+    const { id, setUsersHandler, toggleError } = this.props;
+    const isDeleted = await removeUserData(id);
+    const users = await getAllUsers();
+    if (isDeleted && users) {
+      await setUsersHandler(users);
```

Comment:
could you give it a clear name so I can understand what the method really does?

Reply:
Before this method did `setState` users received from firebase. This method is now removed and the state is updated in componentDidUpdate.

```
  deleteUserHandler = async () => {
    const { selectedUserId, users } = this.state;
    const updatedUsers = users.filter((item) => item.id !== selectedUserId);

    await removeUserData(selectedUserId);
    this.setState({ users: updatedUsers });
  };
```

```
  async componentDidUpdate(prevProps, prevState) {
    const { users, selectedUserId } = this.state;

    if (prevState.users.length > users.length && prevState.users.length) {
      await this.getUsers();
      this.toggleModalDeleteHandler();

      return;
    }
```